src/stateexport.cpp

```diff
@@ -30,7 +30,7 @@ qint64 StateExport::saveVChar64(State* state, QFile& file)
 {
     StateImport::VChar64Header header;
 
-    strncpy(header.id, "VChar", 5);
+    strcpy(header.id, "VChar");
```

Comment:
why this change?
it triggers this error: https://travis-ci.org/ricardoquesada/vchar64/jobs/384181673

Reply:
this change is nonsense - it will copy 6 instead of just 5 bytes - hence the error

Reply:
Sorry for the late reply. Yes in this case strcpy is stupid. Sorry for that, I assumed the target is a char*.

Regardless gcc 8 is stricter towards strncpy:
~~~
../../src/stateexport.cpp:33:12: error: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying 5 bytes from a string of the same length [-Werror=stringop-truncation]
     strncpy(header.id, "VChar", 5);
     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
~~~
I assume it does not like to treat the char[5] as string target but without a termination \0.
I assume you either need to mark the header.id with:

~~~c++
__attribute__((nonstring))
~~~

or fallback to memcpy instead since there is no need for handling \0 termination.

Reply:
thanks. fixed.